cli/tools/check.rs

```diff
@@ -297,163 +310,128 @@ impl TypeChecker {
   }
 }
 
-enum CheckHashResult {
-  Hash(CacheDBHash),
-  NoFiles,
+struct TscRoots {
+  roots: Vec<(ModuleSpecifier, MediaType)>,
+  missing_diagnostics: tsc::Diagnostics,
+  maybe_check_hash: Option<CacheDBHash>,
 }
 
-/// Gets a hash of the inputs for type checking. This can then
-/// be used to tell
-fn get_check_hash(
```

Comment:
Merging this into `get_tsc_roots` is more reliable because then the code is the same. There were some slight discrepancies before.